h/static/scripts/widget-controller.js

```diff
@@ -312,9 +347,11 @@ module.exports = function WidgetController(
     streamer.reconnect();
   });
 
-  // When a direct-linked annotation is successfully anchored in the page,
-  // focus and scroll to it
   $rootScope.$on(events.ANNOTATIONS_SYNCED, function (event, tags) {
```

Comment:
@sheetaluk - Thanks for the clarification. At the moment I think it is difficult to follow what the business rules are from the code. Having two different listeners set up also means we can run into the problem @finitud is currently looking into where a subscription to some event is set up but then not removed when it should be. When the event is fired in future, a listener that should no longer be triggered ends up handling the event.

The way I would approach this is to first enumerate what the rules are (as you've done above) then try to see if we can simplify them and finally try to express them in code as simply as possible.

Looking at these 4 cases, I think (1-3) can be reduced to one. Whenever we reset the currently loaded set of annotations, the selected tab should be switched to the Annotations tab. I think it should be possible to make that happen without these two event listeners:

1. When the app starts, we can set the initial tab to Annotations (in the initial state in `annotationUI`).
2. When we reset the set of loaded annotations, which happens whenever `loadAnnotations()` is called in `WidgetController` (gah, that's such a stupid name for a class), the tab should be set to `Annotations`.

> For cases 1 and 4, after the ANNOTATIONS_SYNCED event is fired, we can tell between anchored and unanchored annotations. 

In the first case, since we always select the same tab, we don't need to care whether the annotations are orphaned or not do we?